pandas/tests/frame/test_operators.py

```diff
@@ -1161,6 +1161,21 @@ def test_inplace_ops_identity(self):
         assert_frame_equal(df2, expected)
         assert df._data is df2._data
 
+        # no id change
+        arr = np.array([1., 2., 3.])
+        df = DataFrame({'a': arr.copy()})
+        expected = id(df)
+        df += 2.
+        assert id(df) == expected
+        df -= 2.
+        assert id(df) == expected
+        df *= 2.
+        assert id(df) == expected
+        df /= 2.
```

Comment:
this fails before your change?

Reply:
yes, it will only fail for python 2.7 using classic division (which people shouldn't be using but the behavior is still unexpected)

you can verify by doing 
```git checkout master pandas/core/ops.py ; pytest pandas/tests/frame/test_operators.py``` 
since the fix and test are in one commit.